staging/src/k8s.io/code-generator/cmd/api-linter/linters/goJSONNameMatch.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Go field names must be CamelCase. JSON field names must be camelCase. Other than capitalization of the initial letter, the two should almost always match. No underscores nor dashes in either.
+// This validator verifies the convention "Other than capitalization of the initial letter, the two should almost always match."
+// Examples (also in unit test):
+//     Go name | JSON name | match
+//     PodSpec   podSpec     true
+//     PodSpec   spec        false
+//     Spec      podSpec     false
+//     JSONSpec  jsonSpec    true
+//     JSONSpec  jsonspec    true
+
+package linters
```

Comment:
I would just call this linter/file namesMatch. Maybe put each linter in their own package?

Maybe we shouldn't call these linters, maybe just "rules" is better. Linters sounds more optional than we want these to be.